### PR TITLE
Use property attribute for opengraph meta tags

### PR DESCRIPTION
**Pull request by jfsoul**
## What does this change?
Uses the `property` attribute in meta tags for opengraph specifically. We believe that is why echobox is using the page title instead of the `og:title` when rendering previews. The facebook opengraph debugger specifically flags that we are not using the property attribute:

image

For extra confidence, this will also match the behaviour of frontend - which has never had these problems.

## Why?
Fixes social shares, particularly ones that rely on opengraph - like hangout chat (see trello card), facebook (see screenshot above) or echobox.

## Link to supporting Trello card
https://trello.com/c/lGkJLv2E/730-implement-open-graph-in-dotcom-rendering-correctly

### CODE REVIEW

**Comment by jfsoul** [comment]
I guess something to flag is that I haven't tested this beyond checking the page source looks good myself, because all the debuggers need to be able to scrape the page AFAIK - so I'd need to do it on CODE.

File: src/web/server/htmlTemplate.ts [hunk 1]
```diff
@@ -84,18 +84,24 @@ export const htmlTemplate = ({
             )}" as="font" crossorigin>`,
     );
 
-    const generateMetaTags = (dataObject: { [key: string]: string }) => {
+    const generateMetaTags = (
+        dataObject: { [key: string]: string },
+        attributeName: 'name' | 'property',
```
Comment (jfsoul) [inline src/web/server/htmlTemplate.ts:89]: not sure whether this is the idiomatic way to do it...

File: src/web/server/document.test.tsx [hunk 2]
```diff
@@ -23,6 +23,33 @@ test('that all the required meta SEO fields exist', () => {
     );
 });
 
+test('that all some of the most important opengraph meta tags exist', () => {
```
Comment (oliverlloyd) [inline src/web/server/document.test.tsx:26]:
```suggestion
test('that the most important opengraph meta tags exist', () => {
```

Comment (oliverlloyd) [reply src/web/server/document.test.tsx:26]: Sorry, a pedantic grammar change!

File: src/web/server/document.test.tsx [hunk 3]
```diff
@@ -23,6 +23,33 @@ test('that all the required meta SEO fields exist', () => {
...
+    ];
+
+    names.map(name =>
+        expect(result).toEqual(
+            expect.stringContaining(`<meta property="${name}"`),
+        ),
+    );
+});
+
+test('that all some of the most important twitter meta tags exist', () => {
```
Comment (oliverlloyd) [inline src/web/server/document.test.tsx:43]:
```suggestion
test('that the most important twitter meta tags exist', () => {
```

File: src/web/server/htmlTemplate.ts (hunk 1 above)
Comment (oliverlloyd) [reply src/web/server/htmlTemplate.ts:89]: Sometimes I upgrade to:

```
const generateMetaTags = ({
        dataObject,
        attributeName,
}: {
        dataObject: { [key: string]: string },
        attributeName: 'name' | 'property',
}) => ...
```

To guard against property order confusion but it's more verbose. I think you can go both ways here.

**Comment by jfsoul** [comment]
> Just a small thought What do you think about splitting generateMetaTags into two functions, generateOGTags & generateTwitterTags? The downside is repetition but it could make things easier to maintain if there are more use cases like this in future where the requirement branch.

Neither here nor there for me. I've tried your suggestion and I'm leaning towards the original for brevity and to hint at the namespace of meta attributes i.e. `name | property` and not necessarily tie that to twitter or og.

File: src/web/server/document.test.tsx [hunk 4]
```diff
@@ -23,6 +23,33 @@ test('that all the required meta SEO fields exist', () => {
     );
 });
 
+test('that the most important opengraph meta tags exist', () => {
```
Comment (gtrufitt) [inline src/web/server/document.test.tsx:26]: 🎊

**Pull request merged**